src/machine_learning/optimization/k_nearest_neighbors.rs

```diff
@@ -0,0 +1,294 @@
+#![allow(dead_code)]
+#![allow(unused_variables)]
+
+use std::{
+    cmp,
+    collections::{BTreeMap, HashMap},
+};
+
+use crate::math::{abs, square_root};
+#[derive(Debug, Clone)]
+pub struct Point {
+    x: f64,        // x-axis
+    y: f64,        // y-axis
+    label: String, // A label denotes the classification of the data. For instance, 'Football' might be labeled as 'Sport'
```

Comment:
These comments do not provide any additional value and can be removed, since.

Why do you restrict yourself to 2D data?

Reply:
Yeah , I agree. I started with 2D data. I updated the code now to support n-dimensions data.